### PR TITLE
Added request for Condition expression

### PR DESCRIPTION
**Pull request by brian-duffy**

### CODE REVIEW

File: src/dynamodb/DynamoDBManager.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export class DynamoDBManager {
       putItemInput.Item = {};
     }
 
+    putItemInput[this.config.conditionExpressionAttributeName] = putPointInput.ConditionExpression;
```
Comment (clakech) [inline src/dynamodb/DynamoDBManager.ts]: why do you setup ConditionExpression in a const var conditionExpressionAttributeName ?
```
putItemInput. ConditionExpression = putPointInput.ConditionExpression;

**Comment by rh389** [comment]
Sorry for the delay, but I think this is a misunderstanding. Anything you want to pass through to the underlying `putItem` call, including `ConditionExpression`, can be achieved by placing it within `putPointInput.PutItemInput` when you call `putPoint`.

```js
myGeoTableManager.putPoint({
        RangeKeyValue: { S: '1234' }, // Use this to ensure uniqueness of the hash/range pairs.
        GeoPoint: { // An object specifying latitutde and longitude as plain numbers. Used to build the geohash, the hashkey and geojson data
            latitude: 51.51,
            longitude: -0.13
        },
        PutItemInput: { // Passed through to the underlying DynamoDB.putItem request. TableName is filled in for you.
            Item: { // The primary key, geohash and geojson data is filled in for you
                country: { S: 'UK' }, // Specify attribute values using { type: value } objects, like the DynamoDB API.
                capital: { S: 'London' }
            }
            ConditionExpression: // [Your condition expression here]
            // Anything else to pass through to `putItem`
        }
    }).promise()
    .then(function() { console.log('Done!') });
```

**Comment by davidlukewilcox** [comment]
@rh389 Sorry, but I don't see where that's being done:

```
  public putPoint(putPointInput: PutPointInput): Request<PutPointOutput, AWSError> {
    const geohash = S2Manager.generateGeohash(putPointInput.GeoPoint);
    const hashKey = S2Manager.generateHashKey(geohash, this.config.hashKeyLength);
    const putItemInput: PutItemInput = {
      TableName: this.config.tableName,
      Item: putPointInput.PutItemInput.Item || {}
    }

    putItemInput.Item[this.config.hashKeyAttributeName] = { N: hashKey.toString(10) };
    putItemInput.Item[this.config.rangeKeyAttributeName] = putPointInput.RangeKeyValue;
    putItemInput.Item[this.config.geohashAttributeName] = { N: geohash.toString(10) };
    putItemInput.Item[this.config.geoJsonAttributeName] = {
      S: JSON.stringify({
        type: 'POINT',
        coordinates: (this.config.longitudeFirst ?
          [putPointInput.GeoPoint.longitude, putPointInput.GeoPoint.latitude] :
          [putPointInput.GeoPoint.latitude, putPointInput.GeoPoint.longitude])
      })
    };

    return this.config.dynamoDBClient.putItem(putItemInput);
  }
```

I see where we take the item wholesale. I don't see where we take other attributes around the item wholesale. I don't see how it's possible to create a `ConditionExpression` attribute on `putItemInput`.

This is functionality that I'd like since I want to ensure that I do not replace any points.

**Comment by rh389** [comment]
@davidlukewilcox you're absolutely right - that's a bug. I'll get on this today.

**Comment by davidlukewilcox** [comment]
Thanks for that... Did you also update batchWritePoints? I don't think that you did. Did you mean to? Does it make sense to update that one as well?

**Pull request closed without merging**